cuegui/cuegui/JobMonitorTree.py

```diff
@@ -259,6 +259,15 @@ def removeFinishedItems(self):
         for item in self.findItems("Finished", QtCore.Qt.MatchFixedString, COLUMN_STATE):
             self.removeItem(item)
 
+    def removeAllJobsExcept(self, keepList):
+        """Removes jobs where the name doesn't match the given search string"""
+        keepKeys = [cuegui.Utils.getObjectKey(job) for job in keepList]
+        existingItems = [self.topLevelItem(index) for index in range(self.topLevelItemCount())]
```

Comment:
We can optimize a bit here by combining these two lines. Something like:
```
for indexNumber in range(self.topLevelItemCount()):
    existingItem = self.topLevelItem(index)
```

Reply:
Done.